x-pack/legacy/plugins/siem/server/lib/detection_engine/alerts/types.ts

```diff
@@ -31,7 +31,7 @@ export interface SignalAlertParams {
   interval: string;
   id: string;
   language: string | undefined;
-  maxSignals: number;
+  maxSignals: number | undefined;
```

Comment:
See my comments above about if this type should change to be an `undefined` here. The source of "truth" for deriving TypeScript is from the alert schema as shown above which allows defaults.